pkgs/by-name/im/imager/package.nix

```diff
@@ -0,0 +1,111 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  gtk2-x11,
+  pkg-config,
+  python3,
+  gfortran,
+  cfitsio,
+  getopt,
+  perl,
+  groff,
+  which,
+  darwin,
+  ncurses,
+  makeWrapper,
+}:
+
+let
+  python3Env = python3.withPackages (
+    ps: with ps; [
+      numpy
+      setuptools
+    ]
+  );
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  version = "4.4-01";
+  pname = "imager";
+
+  src = fetchurl {
+    # The recommended download link is on Nextcloud instance that
+    # requires to accept some general terms of use. Use a mirror at
+    # univ-grenoble-alpes.fr instead.
+    url = "https://cloud.univ-grenoble-alpes.fr/s/J6yEqA6yZ8tX9da/download?path=%2F&files=imager-dec24.tar.gz";
+    hash = "sha256-Pq92IsGY4heekm5zNGngnp6J6YiCHYAyuMT2RsD1/9o=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    groff
+    perl
+    getopt
+    gfortran
+    which
+    makeWrapper
+  ];
+
+  buildInputs = [
+    gtk2-x11
+    cfitsio
+    python3Env
+    ncurses
+  ];
+
+  patches =
+    [
+      ./python-ldflags.patch
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin ([
+      ./clang.patch
+      ./cpp-darwin.patch
+    ]);
+
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-unused-command-line-argument";
+
+  # Workaround for https://github.com/NixOS/nixpkgs/issues/304528
+  env.GAG_CPP = lib.optionalString stdenv.hostPlatform.isDarwin "${gfortran.outPath}/bin/cpp";
+
+  NIX_LDFLAGS = lib.optionalString stdenv.hostPlatform.isDarwin (
+    with darwin.apple_sdk.frameworks; "-F${CoreFoundation}/Library/Frameworks"
+  );
+
+  configurePhase = ''
+    substituteInPlace utilities/main/gag-makedepend.pl --replace-fail '/usr/bin/perl' ${lib.getExe perl}
```

Comment:
patching should be done in postPatch

Reply:
Thanks for the suggestion. Fixed with  d97e9f75fd007de1db8744fcbd47bc4c40cc78f6